awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -337,7 +337,7 @@ def authenticate(self, **kwargs):
         # If we have neither of these, then we can try un-authenticated access
         self.authenticated = True
 
-    def delete_if_needed(self, existing_item, on_delete=None):
+    def delete_if_needed(self, existing_item, on_delete=None, on_continue=None):
```

Comment:
I'm not sure we want to call these `on_continue`. If I am reading this right, on_continue is really telling the function to just not auto-exit for you, is that correct? If so maybe you do something like auto_exit=True and then do something like:
```
if auto_exit:
    self.exit_json(**self.json_output)
```
You don't need to return self.json_output because the calling code can just do `module.json_output` to get at it.

The on_delete if accepting and calling a function if a delete happened, thus its name. On a delete of an object, call this additional function.

Reply:
Agree on the name chnage, and will look at return. Will do the module.json_output as well

Reply:
in terms of self.exit_json, that looks to be the norm for tower_api and is in 10 other places in the plugin. Right now just replicating the norm.